[Fleet server config] IAM authentication for MySQL and Redis (#32219)
fleetdm/fleet: docs/Configuration/fleet-server-configuration.md

    sql_mode: ANSI
  ```

### mysql_sts_assume_role_arn

Amazon Resource Name (ARN) of the AWS Security Token Service (STS) role to assume when using Identity and Access Management (IAM) for MySQL authentication.

- Default value: `""`
- Environment variable: `FLEET_MYSQL_STS_ASSUME_ROLE_ARN`
- Config file format:
  ```yaml
  mysql:
    sts_assume_role_arn: arn:aws:iam::1234567890:role/rds-auth-role
  ```

### mysql_sts_external_id

> AWS Security Token Service (STS) External ID to use for MySQL authentication. Specify this with `mysql_sts_assume_role_arn` to ensure that only the intended AWS account can assume the role.

Author: ```suggestion
Optionally, if you're using a third-party to manage AWS resources, this is the AWS Security Token Service (STS) External ID to use for MySQL authentication. Specify this with `mysql_sts_assume_role_arn`.
```

> Optionally, if you're using a third-party to manage AWS resources, this is the AWS Security Token Service (STS) External ID to use for MySQL authentication. Specify this with `mysql_sts_assume_role_arn`.

Author: ```suggestion
Optionally, if you're using a third-party to manage AWS resources, this is the AWS Security Token Service (STS) External ID to use for IAM authentication. Specify this with `mysql_sts_assume_role_arn`.
```

    sql_mode: ANSI
  ```

### mysql_sts_assume_role_arn

> Amazon Resource Name (ARN) of the AWS Security Token Service (STS) role to use for MySQL authentication. 
> 
> If set, Fleet uses AWS Identity and Access Management (IAM) authentication instead of basic authentication set by `mysql_username` and `mysql_password` or `mysql_password_path`.

Reviewer: @noahtalerman we should update this as well
```suggestion
Amazon Resource Name (ARN) of the AWS Security Token Service (STS) role to assume when using Identity and Access Management (IAM) for MySQL authentication.
```

For the address of the Redis server that Fleet should connect to, include the hostname and port.

> If an AWS ElastiCache endpoint is specified and `redis_password` isn't specified, Identity and Access Management (IAM) authentication is automatically used.

Reviewer: ```suggestion
If an AWS ElastiCache endpoint is specified and `redis_password` isn't specified, Fleet will first attempt to use Identity and Access Management (IAM) authentication before falling back to standard auth without a password.
```

@noahtalerman This relates to your question below about roles as well. If we don't want to add an explicit `redis_use_iam` flag, the only way we can support IAM-less connections without a password is to first _try_ to get a password via IAM, and if that fails, try to connect without a password.

For the `redis_sts_assume_role_arn` config, I would say something like:

> Amazon Resource Name (ARN) of the AWS Security Token Service (STS) role to assume when using Identity and Access Management (IAM) for Redis authentication.

    use_tls: true
  ```

### redis_sts_assume_role_arn

Amazon Resource Name (ARN) of the AWS Security Token Service (STS) role to use for Redis authentication. 

> If set, Fleet uses AWS Identity and Access Management (IAM) authentication instead of basic authentication set by `redis_username` and `redis_password` (optional).

Reviewer: See comment above re: connecting without a password.
